store/tikv/pd_codec.go

```diff
@@ -26,25 +26,28 @@ type codecPDClient struct {
 
 // GetRegion encodes the key before send requests to pd-server and decodes the
 // returned StartKey && EndKey from pd-server.
-func (c *codecPDClient) GetRegion(key []byte) (*metapb.Region, error) {
+func (c *codecPDClient) GetRegion(key []byte) (*metapb.Region, *metapb.Peer, error) {
 	encodedKey := codec.EncodeBytes([]byte(nil), key)
-	region, err := c.Client.GetRegion(encodedKey)
+	region, peer, err := c.Client.GetRegion(encodedKey)
```

Comment:
I suppose the peer maybe nil if tikv does not sync leader to pd in time?
So I didn't treat this situation as an error and handled it at https://github.com/pingcap/tidb/pull/1456/files#diff-708f6242b27e2b7bcf0e905e9b0eacf2R216